feature/rds/auth/builder_test.go

```diff
@@ -0,0 +1,59 @@
+package auth_test
+
+import (
+	"context"
+	"net/url"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/rds/auth"
+)
+
+func TestConnectionStringBuilder(t *testing.T) {
+	cases := []struct {
+		user     string
+		endpoint string
+		region   string
+		dbName   string
+		values   url.Values
+		format   auth.ConnectionFormat
+		creds    aws.CredentialsProvider
+
+		expectErr            bool
+		expectedConnectRegex string
+	}{
+		{
+			user:                 "foo",
+			endpoint:             "foo.bar",
+			region:               "region",
+			dbName:               "name",
+			format:               auth.NoConnectionFormat,
+			creds:                &staticCredentials{AccessKey: "AKID", SecretKey: "SECRET", Session: "SESSION"},
+			expectErr:            true,
+			expectedConnectRegex: "",
+		},
```

Comment:
This test will be removed/change if `NoConnectionFormat` is not kept as a valid enum